modules/apps/frontend-token/frontend-token-definition-test/src/testIntegration/java/com/liferay/frontend/token/definition/test/FrontendTokenDefinitionRegistryTest.java

```diff
@@ -0,0 +1,228 @@
+import com.liferay.portal.kernel.service.LayoutSetLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.URLUtil;
+import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Anderson Luiz
+ * @author Thiago Buarque
+ */
+@FeatureFlags("LPD-10773")
+@RunWith(Arquillian.class)
+public class FrontendTokenDefinitionRegistryTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_frontendTokenDefinitionJSONString = JSONFactoryUtil.createJSONObject(
+			URLUtil.toString(
+				FrontendTokenDefinitionRegistryTest.class.getResource(
+					"dependencies/frontend-token-definition.json"))
+		).toString();
+
+		Group group = GroupTestUtil.addGroup();
+
+		_layoutSet = _layoutSetLocalService.fetchLayoutSet(
+			group.getGroupId(), false);
+
+		User user = UserTestUtil.addUser();
+
+		_clientExtensionEntry =
+			_clientExtensionEntryLocalService.addClientExtensionEntry(
+				"any-external-reference", user.getUserId(), "",
+				HashMapBuilder.put(
+					LocaleUtil.getDefault(), "Client Extension Name"
+				).build(),
+				"", "", ClientExtensionEntryConstants.TYPE_THEME_CSS,
+				"frontendTokenDefinition=" +
+					_frontendTokenDefinitionJSONString);
+
+		_clientExtensionEntryRel =
+			_clientExtensionEntryRelLocalService.addClientExtensionEntryRel(
+				user.getUserId(), _layoutSet.getGroupId(),
+				_portal.getClassNameId(LayoutSet.class),
+				_layoutSet.getLayoutSetId(),
+				_clientExtensionEntry.getExternalReferenceCode(),
+				_clientExtensionEntry.getType(), StringPool.BLANK,
+				ServiceContextTestUtil.getServiceContext(
+					group.getGroupId(), TestPropsValues.getUserId()));
+	}
+
+	@After
+	public void tearDown() throws PortalException {
+		_clientExtensionEntryLocalService.deleteClientExtensionEntry(
+			_clientExtensionEntry.getClientExtensionEntryId());
+
+		_clientExtensionEntryRelLocalService.createClientExtensionEntryRel(
+			_clientExtensionEntryRel.getClientExtensionEntryRelId());
+	}
+
+	@Test
+	public void testGetClientExtensionFrontendTokenDefinition()
```

Comment:
How does this test differ from `testGetWorkspaceFrontendTokenDefinition`? They have the same code... :thinking:

Reply:
Oh, that was my bad. I refactored to use the `LayoutSet` but missed passing the right LayoutSet for each case